app/core/utils.py

```diff
@@ -196,3 +196,33 @@ def get_gpu_info():
     mem_percentage = (used_memory_value / total_memory_value) * 100
 
     return gpu_name, total_memory_value, used_memory_value, mem_percentage
+
+
+cached_domain = None
+
+
+def check_dns_exists():
+    global cached_domain
+
+    if cached_domain is not None:
+        return cached_domain
+
+    url = "http://dnsd:8080/dns/existing"
```

Comment:
Same as above. All the environment variables and constants should go in the `config.py` file.

Reply:
Done